doc/src/build/programming-with-objects/ch4-object-wrapping.md

```diff
@@ -0,0 +1,195 @@
+## Chapter 4: Object Wrapping
+In most languages, we organize data structures in layer by nesting complex data structures in another data structure. In Move, you could do the same by putting a field of struct type in another, like the following:
+```rust
+struct Foo has key {
+    id: VersionedID,
+    bar: Bar,
+}
+struct Bar has store {
+    value: u64,
+}
+```
+> :bulb: For a struct type to be capable of being embedded in a Sui object struct (which will have `key` ability), the embedded struct type must have `store` ability.
+
+In the above example, `Bar` is a normal Move struct that cannot be used to construct Sui objects, since it doesn't have `key` ability. This is very common usage when we just need to organize data with good encapsulation.
+In some cases, however, we want to put a Sui object struct type as a field in another Sui object struct type. In the above example, if we change `Bar` into:
+```rust
+struct Bar has key, store {
+    id: VersionedID,
+    value: u64,
+}
+```
+Now `Bar` is also a Sui object type. When we put a Sui object of type `Bar` into a Sui object of type `Foo`, the Sui object of type `Bar` is said to be **wrapped**.
+There are some interesting consequences of wrapping an Sui object into another. When an object is wrapped, this object no longer exists independently on-chain. One will no longer be able to look up this object by its ID. This object becomes part of the data of the object that wraps it. Most importantly, one can no longer pass the wrapped object as an argument in any way in Move calls. The only access point is through the wrapping object.
```

Comment:
Do should probably explain 3 edge cases (probably with a note):

1. It's not clear if we can wrap an already wrapped object?
wrap(wrap(wrap(obj)))

2. Are accidental cycles resolved automatically at compilation or the dev should take care of it?

3. Can a regular struct hold a Sui object (thus the opposite from the example)?

Reply:
Good call! Will explain them in the doc.
To answer your question
1) No. A wrapped object cannot be used directly.
2) Since 1) is no, there cannot be cycles
3) No.